### PR TITLE
feature: support defined environment variable in config.yaml #1922

### PR DESCRIPTION
**Pull request by moorefu**
### What this PR does / why we need it:
implementation #1922
user can inject customer environment variable by modify config.yaml file. it is useful in some case, for example
1. use oci to read oracle dbms, require TNS_ADMIN to inject tnsnames.ora dir.
2.etc

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible?

### CODE REVIEW

**Comment by membphis** [comment]
please add test case for this feature

**Comment by moorefu** [comment]
> please add test case for this feature

yes i add one test case in  .travis/apisix_cli_test.sh . is it right?

File: .travis/apisix_cli_test.sh [hunk 1]
```diff
@@ -161,3 +161,21 @@ if [ ! $? -eq 0 ]; then
...
+set +ex
+ed ":a;N;s/\n//g;ta" conf/nginx.conf |grep -E "env TNS_ADMIN;env DATACENTER_ID=1;" > /dev/null
+if [ ! $? -eq 0 ]; then
+    echo "failed: custom environment variable in nginx.conf is not expect"
+    exit 1
+fi
+
+set -ex
+
+echo "passed: custom environment variable in nginx.conf is ok"
```
Comment (membphis) [inline .travis/apisix_cli_test.sh]: please add a blank line at the end of the file.

**Comment by membphis** [comment]
@moorefu you need to rebase your branch first

**Comment by spacewander** [comment]
The feature is implemented in #2174.

**Pull request closed without merging**